bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -950,19 +939,21 @@ public long addEntry(Long ledger, ByteBuf entry, boolean rollLog) throws IOExcep
 
             /**
              * if disk of the logChannel is full or if the entrylog limit is
-             * reached of if the logchannel is not initialized, then
+             * reached or if the logchannel is not initialized, then
              * createNewLog. If allDisks are full then proceed with the current
              * logChannel, since Bookie must have turned to readonly mode and
              * the addEntry traffic would be from GC and it is ok to proceed in
              * this case.
              */
             if ((diskFull && (!allDisksFull)) || reachEntryLogLimit || (logChannel == null)) {
-                flushCurrentLog(logChannel, false);
+                if (logChannel != null) {
+                    logChannel.flushAndForceWriteIfRegularFlush(false);
```

Comment:
this is added because with this commit https://github.com/apache/bookkeeper/commit/30261eae3fd8ab25239d57cfb86a200d5f7b6b7d#diff-66d58c001864b142c348f9191407c410 it was deleted and it was a mistake.